File: python/problem1481.py
class Solution:
    def findLeastNumOfUniqueInts(self, arr: List[int], k: int) -> int:
        count = collections.defaultdict(int)
        for n in arr:
            count[n] += 1
        values = list(count.values())
        values.sort()
        removeCount = 0
        for v in values:
            if v <= k:
                k -= v
                removeCount += 1
            else:
                break
        return len(values) - removeCount

File: Java/problem1481.java
class Solution {
    public int findLeastNumOfUniqueInts(int[] arr, int k) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int a : arr) {
            map.put(a, map.getOrDefault(a, 0) + 1);
        }
        int[] values = new int[map.size()];
        int i = 0;
        for (int v : map.values()) {
            values[i++] = v;
        }
        Arrays.sort(values);
        int count = 0;
        for (i = 0; i < values.length; ++i) {
            if (values[i] <= k) {
                ++count;
                k -= values[i];
            }
        }
        return values.length - count;
    }
}
